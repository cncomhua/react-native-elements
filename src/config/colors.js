import { Platform, StyleSheet } from 'react-native';

export default {
  primary: Platform.select({
    default: '#2196f3', // rgb(33, 150, 243) Android
    ios: '#007aff' // rgb(0, 122, 255)
  }),
  secondary: '#8F0CE8',
  grey0: '#393e42',
  grey1: '#43484d',
  grey2: '#5e6977',
  grey3: '#86939e',
  grey4: '#bdc6cf',
  grey5: '#e1e8ee',
  greyOutline: '#bbb',
  searchBg: '#303337',
  success: Platform.select({
    default: '#4caf50', // Android
    ios: '#4cd964'
  }),
  error: Platform.select({
    default: '#f44336', // Android
    ios: '#ff3b30'
  }),
  warning: Platform.select({
    default: '#ffeb3b', // Android
    ios: '#ffcc00'
  }),
  disabled: 'hsl(208, 8%, 90%)',
  // Darker color if hairlineWidth is not thin enough
  divider: StyleSheet.hairlineWidth < 1 ? '#bcbbc1' : 'rgba(0, 0, 0, 0.12)',
};
