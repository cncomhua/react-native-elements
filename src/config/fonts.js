export default {
  android: {
    regular: {
      fontFamily: 'sans-serif',
      fontWeight: 'regular',
    },
    light: {
      fontFamily: 'sans-serif-light',
      fontWeight: 'regular',
    },
    condensed: {
      fontFamily: 'sans-serif-condensed',
      fontWeight: 'regular',
    },
    condensed_light: {
      fontFamily: 'sans-serif-condensed',
      fontWeight: 'light',
    },
    black: {
      // note(brentvatne): sans-serif-black is only supported on Android 5+,
      // we can detect that here and use it in that case at some point.
      fontFamily: 'sans-serif',
      fontWeight: 'bold',
    },
    thin: {
      fontFamily: 'sans-serif-thin',
      fontWeight: 'regular',
    },
    medium: {
      fontFamily: 'sans-serif-medium',
      fontWeight: 'regular',
    },
    bold: {
      fontFamily: 'sans-serif',
      fontWeight: 'bold',
    },
  },
  default: {},
};
