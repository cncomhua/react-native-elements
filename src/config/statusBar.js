import { Dimensions, Platform, StatusBar } from 'react-native';

// Borrowed from https://github.com/ptelad/react-native-iphone-x-helper

function isIphoneX() {
  const dimen = Dimensions.get('window');
  return (
    Platform.OS === 'ios' &&
    !Platform.isPad &&
    !Platform.isTVOS &&
    (dimen.height === 812 || dimen.width === 812)
  );
}

function getIphoneStatusBarHeight() {
  if (isIphoneX()) {
    return 44;
  }
  return 20;
}

function getAndroidStatusBarHeight(skip) {
  if (skip) {
    return 0;
  }
  return StatusBar.currentHeight;
}

export function getStatusBarHeight(skipAndroid = false) {
  return Platform.select({
    ios: getIphoneStatusBarHeight(),
    android: getAndroidStatusBarHeight(skipAndroid),
    default: 0,
  });
}
