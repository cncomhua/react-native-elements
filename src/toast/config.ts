import { StylePropsWithMessageType, ToastPosition } from './ToastProvider';

import type { StyleProp, TextProps, TextStyle, ViewStyle } from 'react-native';

export type DefaultConfigType = {
  duration: number;
  maxMessages: number;
  position: ToastPosition;
  containerToastStyle: StyleProp<ViewStyle>;
  containerMessageStyle: StylePropsWithMessageType<ViewStyle>;
  textMessageStyle: StylePropsWithMessageType<TextStyle>;
  textMessageProps: TextProps;
};

const defaultConfig: DefaultConfigType = {
  duration: 2000,
  maxMessages: 5,
  position: ToastPosition.top,
  containerToastStyle: {
    position: 'absolute',
    left: 0,
    right: 0,
    zIndex: 99999,
  },
  containerMessageStyle: {
    margin: 10,
    marginBottom: 5,
    padding: 10,
    borderRadius: 4,
    backgroundColor: '#586ae3',
    shadowColor: 'black',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.15,
    shadowRadius: 5,
    elevation: 6,
    // typed styles
    info: {},
    error: {},
    warning: {},
    success: {},
  },
  textMessageStyle: {
    // typed styles
    info: {},
    error: {},
    warning: {},
    success: {},
  },
  textMessageProps: {},
};

export default defaultConfig;
