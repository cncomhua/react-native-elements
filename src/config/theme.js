// For testing purposes only
import colors from './colors';

export default {
  colors,
};
