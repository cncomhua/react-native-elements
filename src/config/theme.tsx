import { ViewProps } from 'react-native';
import { SwipeRatingProps } from '../Rating';
import { TapRatingProps } from '../AirbnbRating';
import { AvatarProps, AccessoryProps } from '../Avatar';
import { BadgeProps } from '../Badge';
import { BottomSheetProps } from '../BottomSheet';
import { ButtonProps } from '../Button';
import { ButtonGroupProps } from '../ButtonGroup';
import { CardProps } from '../card/Card';
import { CheckBoxProps } from '../Checkbox';
import { DividerProps } from '../Divider';
import { HeaderProps } from '../Header';
import { IconProps } from '../Icon';
import { ImageProps } from '../Image';
import { InputProps } from '../Input';
import { ListItemProps } from '../ListItem';
import { ListItemAccordionProps } from '../ListItem';
import { OverlayProps } from '../Overlay';
import { PricingCardProps } from '../PricingCard';
import { SearchBarProps } from '../searchbar/SearchBar';
import { SliderProps } from '../Slider';
import { SocialIconProps } from '../SocialIcon';
import { TextProps } from '../Text';
import { TileProps } from '../Tile';
import { TooltipProps } from '../Tooltip';
import { SwitchProps } from '../Switch';
import { TabItemProps, TabProps } from '../tab/Tab';
import { TabViewProps } from '../tab/TabView';
import { FABProps } from '../FAB';
import { SpeedDialProps, SpeedDialActionProps } from '../SpeedDial';
import { LinearProgressProps } from '../LinearProgress';
import { ChipProps } from '../Chip';
import {
  DialogActionsProps,
  DialogLoadingProps,
  DialogProps,
  DialogTitleProps,
} from '../Dialog';

import colors, { Colors } from './colors';

export default {
  colors,
};

type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };

export interface FullTheme {
  Avatar: Partial<AvatarProps>;
  AvatarAccessory: Partial<AccessoryProps>;
  Badge: Partial<BadgeProps>;
  BottomSheet: Partial<BottomSheetProps>;
  Button: Partial<ButtonProps>;
  ButtonGroup: Partial<ButtonGroupProps>;
  Chip: Partial<ChipProps>;
  Card: Partial<CardProps>;
  CardDivider: Partial<DividerProps>;
  CardFeaturedSubtitle: Partial<TextProps>;
  CardFeaturedTitle: Partial<TextProps>;
  CardImage: Partial<ImageProps>;
  CardTitle: Partial<TextProps>;
  CheckBox: Partial<CheckBoxProps>;
  Divider: Partial<DividerProps>;
  Dialog: Partial<DialogProps>;
  DialogActions: Partial<DialogActionsProps>;
  DialogButton: Partial<ButtonProps>;
  DialogLoading: Partial<DialogLoadingProps>;
  DialogTitle: Partial<DialogTitleProps>;
  Header: Partial<HeaderProps>;
  Icon: Partial<IconProps>;
  Image: Partial<ImageProps>;
  Input: Partial<InputProps>;
  ListItem: Partial<ListItemProps>;
  ListItemAccordion: Partial<ListItemAccordionProps>;
  ListItemButtonGroup: Partial<ButtonGroupProps>;
  ListItemCheckBox: Partial<CheckBoxProps>;
  ListItemContent: Partial<ViewProps>;
  ListItemChevron: Partial<IconProps>;
  ListItemInput: Partial<InputProps>;
  ListItemSubtitle: Partial<TextProps>;
  ListItemTitle: Partial<TextProps>;
  Overlay: Partial<OverlayProps>;
  PricingCard: Partial<PricingCardProps>;
  Rating: Partial<SwipeRatingProps>;
  AirbnbRating: Partial<TapRatingProps>;
  SearchBar: Partial<SearchBarProps>;
  Slider: Partial<SliderProps>;
  SocialIcon: Partial<SocialIconProps>;
  Text: Partial<TextProps>;
  Tile: Partial<TileProps>;
  Switch: Partial<SwitchProps>;
  Tooltip: Partial<TooltipProps>;
  colors: RecursivePartial<Colors>;
  Tab: Partial<TabProps>;
  TabItem: Partial<TabItemProps>;
  TabView: Partial<TabViewProps>;
  TabViewItem: Partial<ViewProps>;
  FAB: Partial<FABProps>;
  SpeedDial: Partial<SpeedDialProps>;
  SpeedDialAction: Partial<SpeedDialActionProps>;
  LinearProgress: Partial<LinearProgressProps>;
}

export type Theme<T = {}> = Partial<FullTheme> & T;

export type UpdateTheme = (updates: RecursivePartial<FullTheme>) => void;

export type ReplaceTheme = (updates: RecursivePartial<FullTheme>) => void;
