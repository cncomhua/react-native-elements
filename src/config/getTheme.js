import { StyleSheet, Platform } from 'react-native';
import base from './colors';
import fonts from './fonts';
import normalize from '../helpers/normalizeText';

export default function getTheme(theme) {
  const colors = {
    ...base,
    ...theme,
  };

  return {
    colors,
    badge: StyleSheet.create({
      container: {
        flexDirection: 'row',
      },
      badge: {
        padding: 12,
        paddingTop: 3,
        paddingBottom: 3,
        backgroundColor: colors.grey1,
        borderRadius: 20,
        alignItems: 'center',
        justifyContent: 'center',
      },
      text: {
        fontSize: 14,
        color: 'white',
      },
    }),
    card: StyleSheet.create({
      container: {
        backgroundColor: 'white',
        borderColor: colors.grey5,
        borderWidth: 1,
        padding: 15,
        margin: 15,
        marginBottom: 0,
        ...Platform.select({
          ios: {
            shadowColor: 'rgba(0,0,0, .2)',
            shadowOffset: { height: 0, width: 0 },
            shadowOpacity: 1,
            shadowRadius: 1,
          },
          android: {
            elevation: 1,
          },
        }),
      },
      featuredTitle: {
        fontSize: normalize(18),
        marginBottom: 8,
        color: 'white',
        ...Platform.select({
          ios: {
            fontWeight: '800',
          },
          android: {
            ...fonts.android.black,
          },
        }),
      },
      featuredSubtitle: {
        fontSize: normalize(13),
        marginBottom: 8,
        color: 'white',
        ...Platform.select({
          ios: {
            fontWeight: '400',
          },
          android: {
            ...fonts.android.black,
          },
        }),
      },
      wrapper: {
        backgroundColor: 'transparent',
      },
      divider: {
        marginBottom: 15,
      },
      cardTitle: {
        fontSize: normalize(14),
        ...Platform.select({
          ios: {
            fontWeight: 'bold',
          },
          android: {
            ...fonts.android.black,
          },
        }),
        textAlign: 'center',
        marginBottom: 15,
        color: colors.grey1,
      },
      imageCardTitle: {
        marginTop: 15,
      },
      overlayContainer: {
        flex: 1,
        alignItems: 'center',
        backgroundColor: 'rgba(0, 0, 0, 0.2)',
        alignSelf: 'stretch',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
      },
    }),
    checkbox: StyleSheet.create({
      wrapper: {
        flexDirection: 'row',
        alignItems: 'center',
      },
      container: {
        margin: 5,
        marginLeft: 10,
        marginRight: 10,
        backgroundColor: '#fafafa',
        borderColor: '#ededed',
        borderWidth: 1,
        padding: 10,
        borderRadius: 3,
      },
      text: {
        marginLeft: 10,
        marginRight: 10,
        color: colors.grey1,
        ...Platform.select({
          ios: {
            fontWeight: 'bold',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
    }),
    divider: StyleSheet.create({
      container: {
        height: 1,
        backgroundColor: colors.grey5,
      },
    }),
    header: StyleSheet.create({
      innerContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'flex-end',
      },
      outerContainer: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        backgroundColor: '#fff',
        borderBottomColor: '#f2f2f2',
        borderBottomWidth: 1,
        padding: 15,
        height: 70,
      },
    }),
    icon: StyleSheet.create({
      button: {
        margin: 7,
      },
      raised: {
        ...Platform.select({
          ios: {
            shadowColor: 'rgba(0,0,0, .4)',
            shadowOffset: { height: 1, width: 1 },
            shadowOpacity: 1,
            shadowRadius: 1,
          },
          android: {
            elevation: 2,
          },
        }),
      },
    }),
    input: StyleSheet.create({
      container: {
        flexDirection: 'row',
        borderBottomWidth: 1,
        borderColor: 'rgba(171, 189, 219, 1)',
        alignItems: 'center',
      },
      iconContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        marginLeft: 15,
      },
      input: {
        alignSelf: 'center',
        color: 'black',
        fontSize: 18,
        marginLeft: 10,
      },
      error: {
        color: '#FF2D00',
        margin: 5,
        fontSize: 12,
      },
    }),
    list: StyleSheet.create({
      listContainer: {
        marginTop: 20,
        borderTopWidth: 1,
        borderColor: colors.greyOutline,
        backgroundColor: colors.white,
      },
    }),
    listItem: StyleSheet.create({
      avatar: {
        width: 34,
        height: 34,
      },
      container: {
        paddingTop: 10,
        paddingRight: 10,
        paddingBottom: 10,
        borderBottomColor: colors.greyOutline,
        borderBottomWidth: 1,
        backgroundColor: 'transparent',
      },
      wrapper: {
        flexDirection: 'row',
        marginLeft: 10,
      },
      iconStyle: {
        justifyContent: 'center',
        alignItems: 'center',
      },
      icon: {
        marginRight: 8,
      },
      title: {
        fontSize: normalize(14),
        color: colors.grey1,
      },
      subtitle: {
        color: colors.grey3,
        fontSize: normalize(12),
        marginTop: 1,
        ...Platform.select({
          ios: {
            fontWeight: '600',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
      titleSubtitleContainer: {
        justifyContent: 'center',
        flex: 1,
      },
      chevronContainer: {
        flex: 0.15,
        alignItems: 'flex-end',
        justifyContent: 'center',
      },
      switchContainer: {
        flex: 0.15,
        alignItems: 'flex-end',
        justifyContent: 'center',
        marginRight: 5,
      },
      rightTitleContainer: {
        flex: 1,
        alignItems: 'flex-end',
        justifyContent: 'center',
      },
      rightTitleStyle: {
        marginRight: 5,
        color: colors.grey4,
      },
      textInputStyle: {
        height: 20,
        textAlign: 'right',
      },
    }),
    pricingCard: StyleSheet.create({
      container: {
        margin: 15,
        marginBottom: 15,
        backgroundColor: 'white',
        borderColor: colors.grey5,
        borderWidth: 1,
        padding: 15,
        ...Platform.select({
          ios: {
            shadowColor: 'rgba(0,0,0, .2)',
            shadowOffset: { height: 1, width: 0 },
            shadowOpacity: 0.5,
            shadowRadius: 0.5,
          },
          android: {
            elevation: 1,
          },
        }),
      },
      wrapper: {
        backgroundColor: 'transparent',
      },
      pricingTitle: {
        textAlign: 'center',
        color: colors.primary,
        fontSize: normalize(30),
        ...Platform.select({
          ios: {
            fontWeight: '800',
          },
          android: {
            ...fonts.android.black,
          },
        }),
      },
      pricingPrice: {
        textAlign: 'center',
        marginTop: 10,
        marginBottom: 10,
        fontSize: normalize(40),
        ...Platform.select({
          ios: {
            fontWeight: '700',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
      pricingInfo: {
        textAlign: 'center',
        marginTop: 5,
        marginBottom: 5,
        color: colors.grey3,
        ...Platform.select({
          ios: {
            fontWeight: '600',
          },
          android: {
            ...fonts.android.bold,
          },
        }),
      },
      button: {
        marginTop: 15,
        marginBottom: 10,
      },
    }),
    search: StyleSheet.create({
      container: {
        borderTopWidth: 1,
        borderBottomWidth: 1,
        borderBottomColor: '#000',
        borderTopColor: '#000',
        backgroundColor: colors.grey0,
      },
      containerLight: {
        backgroundColor: colors.grey5,
        borderTopColor: '#e1e1e1',
        borderBottomColor: '#e1e1e1',
      },
      icon: {
        backgroundColor: 'transparent',
        position: 'absolute',
        top: 15.5,
        ...Platform.select({
          android: {
            top: 20,
          },
        }),
      },
      loadingIcon: {
        backgroundColor: 'transparent',
        position: 'absolute',
        right: 16,
        top: 13,
        ...Platform.select({
          android: {
            top: 18,
          },
        }),
      },
      input: {
        paddingLeft: 26,
        paddingRight: 19,
        margin: 8,
        borderRadius: 3,
        overflow: 'hidden',
        backgroundColor: colors.searchBg,
        fontSize: normalize(14),
        color: colors.grey3,
        height: 40,
        ...Platform.select({
          ios: {
            height: 30,
          },
          android: {
            borderWidth: 0,
          },
        }),
      },
      inputLight: {
        backgroundColor: colors.grey4,
      },
      searchIcon: {
        left: 16,
      },
      clearIcon: {
        right: 16,
      },
    }),
  };
}
