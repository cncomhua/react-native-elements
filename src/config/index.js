import BackgroundImage from './BackgroundImage';
import colors from './colors';
import ThemeProvider from './ThemeProvider';
import getTheme from './getTheme';
import ViewPropTypes from './ViewPropTypes';
import fonts from './fonts';

export {
  BackgroundImage,
  colors,
  fonts,
  ThemeProvider,
  getTheme,
  ViewPropTypes,
};
