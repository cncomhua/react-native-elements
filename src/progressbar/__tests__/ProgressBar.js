import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '../../config';
import ProgressBarThemed, {
  ProgressBarElement as ProgressBar,
} from '../ProgressBar';

describe('ProgressBar Component', () => {
  it('should render without issues', () => {
    const component = shallow(<ProgressBar />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have indeterminate false as default', () => {
    const component = shallow(<ProgressBar />);

    const indeterminate = component.props().indeterminate;
    expect(indeterminate).toBe(false);
  });

  it('should have indeterminate equal to true', () => {
    const component = shallow(<ProgressBar indeterminate />);

    const indeterminate = component.props().indeterminate;
    expect(indeterminate).toBe(true);
  });

  it('should have progress equal to 0.5', () => {
    const component = shallow(<ProgressBar progress={0.5} />);

    const progress = component.props().progress;
    expect(progress).toBe(0.5);
  });

  it('should have props color', () => {
    const theme = {
      ProgressBar: {
        progress: 0.5,
      },
    };

    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <ProgressBarThemed color="red" />
      </ThemeProvider>
    );

    expect(
      component.root.findByType(ProgressBarThemed).children[0].children[0].props
        .color
    ).toBe('red');
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should use values from the theme', () => {
    const theme = {
      ProgressBar: {
        progress: 0.5,
        borderRadius: 16,
      },
    };

    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <ProgressBarThemed />
      </ThemeProvider>
    );

    expect(
      component.root.findByType(ProgressBarThemed).children[0].children[0].props
        .progress
    ).toBe(0.5);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('local props should override style props on theme', () => {
    const theme = {
      ProgressBar: {
        style: {
          borderRadius: 16,
        },
      },
    };

    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <ProgressBarThemed />
      </ThemeProvider>
    );

    expect(
      component.root.findByType(ProgressBarThemed).children[0].children[0].props
        .style.borderRadius
    ).toBe(16);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
