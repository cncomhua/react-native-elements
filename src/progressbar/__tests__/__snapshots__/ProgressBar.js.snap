// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component local props should override style props on theme 1`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  indeterminateAnimationDuration={1000}
  onLayout={[Function]}
  progress={0}
  replaceTheme={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "#2089dc",
        "borderRadius": 0,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": undefined,
      },
      Object {
        "borderRadius": 16,
      },
    ]
  }
  updateTheme={[Function]}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#2089dc",
        "height": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": 1,
          },
          Object {
            "scaleX": 0.0001,
          },
        ],
      }
    }
  />
</View>
`;

exports[`ProgressBar Component should have props color 1`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  indeterminateAnimationDuration={1000}
  onLayout={[Function]}
  progress={0.5}
  replaceTheme={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "red",
        "borderRadius": 0,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": undefined,
      },
      undefined,
    ]
  }
  updateTheme={[Function]}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": 0.5,
          },
          Object {
            "scaleX": 0.50005,
          },
        ],
      }
    }
  />
</View>
`;

exports[`ProgressBar Component should render without issues 1`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  indeterminateAnimationDuration={1000}
  onLayout={[Function]}
  progress={0}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "blue",
        "borderRadius": 0,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": undefined,
      },
      undefined,
    ]
  }
  useNativeDriver={false}
>
  <AnimatedComponent
    style={
      Object {
        "backgroundColor": "blue",
        "height": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": 1,
          },
          Object {
            "scaleX": 0.0001,
          },
        ],
      }
    }
  />
</View>
`;

exports[`ProgressBar Component should use values from the theme 1`] = `
<View
  animated={true}
  animationConfig={
    Object {
      "bounciness": 0,
    }
  }
  animationType="spring"
  indeterminate={false}
  indeterminateAnimationDuration={1000}
  onLayout={[Function]}
  progress={0.5}
  replaceTheme={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderColor": "#2089dc",
        "borderRadius": 16,
        "borderWidth": 1,
        "overflow": "hidden",
        "width": undefined,
      },
      undefined,
    ]
  }
  updateTheme={[Function]}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "backgroundColor": "#2089dc",
        "height": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateX": 0.5,
          },
          Object {
            "scaleX": 0.50005,
          },
        ],
      }
    }
  />
</View>
`;
